src/commands/utils/uploadUDF.test.ts

```diff
@@ -271,24 +283,164 @@ describe("uploadUDF", () => {
     });
 
     it("should handle upload errors properly", async () => {
-      const mockParams = {
-        environment: "env-123456",
-        cloud: "Azure",
-        region: "australiaeast",
-        artifactName: "test-artifact",
-        fileFormat: "jar",
-        selectedFile: tempJarUri,
-      };
+      // Simulate file not found error for the upload
+      const uploadError = new ResponseError(
+        500,
+        "Internal Server Error",
+        new Response(null, { status: 500 }),
+      );
 
-      const uploadError = new Error("Azure upload failed: 500 Internal Server Error");
       uploadFileToAzureStub.rejects(uploadError);
 
+      if (!fs.existsSync(tempJarPath)) {
```

Comment:
Why would it not ever exist? Top scope's beforeEach() will have set up for us, guaranteed.